### PR TITLE
Add local notifications for iOS

### PR DESCRIPTION
**Pull request by s-fletcher**
I was originally going to use [react-native-notifications](https://wix.github.io/react-native-notifications/docs/getting-started), however I was unsuccessful in getting that to work. This is still installed on the [`local-notifications`](https://github.com/tailwind/kite/tree/local-notifications) branch.

Instead, I decided to use [react-native-push-notification](https://www.npmjs.com/package/react-native-push-notification) which was working properly.

demo-notifications
<details>
<summary>File System Diff</summary>

```diff
@@ -102,12 +102,16 @@
   |  |  |-theme.d.ts
   |  |-state
   |  |  |-authSlice.ts
   |  |  |-index.ts
   |  |  |-todoSlice.ts
   |  |-services
+  |  |  |-notificationService
+  |  |  |  |-sendNotification.ts
+  |  |  |  |-registerNotifications.ts
+  |  |  |  |-index.ts
   |  |  |-exampleService
   |  |  |  |-authMock.ts
   |  |  |  |-getExample.ts
   |  |  |  |-client.ts
   |  |  |  |-types.ts
   |  |  |  |-index.ts
```
</details>

### CODE REVIEW

File: ios/kite.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -496,7 +498,7 @@
 					"-ObjC",
 					"-lc++",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = "org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)";
+				PRODUCT_BUNDLE_IDENTIFIER = org.reactjs.native.sam.kite;
```
Comment (murtyjones) [inline ios/kite.xcodeproj/project.pbxproj]: just double checking that it's intentional that this uses your name

**Review by murtyjones — COMMENTED** [review]
I'm not familiar with this stuff so just had a naive question but looks good from a sanity check

File: ios/kite.xcodeproj/project.pbxproj (hunk 1 above)
Comment (s-fletcher) [reply ios/kite.xcodeproj/project.pbxproj]: Nope, it's not. Thanks for catching that. For some reason I needed to change these settings to run it on my iPhone.

**Review by garrettjoecox — COMMENTED** [review]
Seems like a good PoC/Starting point, the notification libraries should be hidden behind a service, that same service should likely store & keep track of the things it has scheduled to fire

File: index.js [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
+import PushNotificationIOS from "@react-native-community/push-notification-ios";
 import React from 'react';
 import { AppRegistry, Platform, UIManager } from 'react-native';
+import PushNotification from "react-native-push-notification";
 import { enableScreens } from 'react-native-screens';
 import { name as appName } from './app.json';
 import { config } from './src/config';
 import { App } from './src/domains/core/App';
 
+PushNotification.configure({
```
Comment (garrettjoecox) [inline index.js]: This logic probably belongs within a hook or service call, invoked from `App.tsx`

Comment (s-fletcher) [reply index.js]: I agree about the hook/service call. 

As for where this should be called, the documentation says to not call it inside a component or App. https://www.npmjs.com/package/react-native-push-notification#usage

Comment (garrettjoecox) [reply index.js]: Eek, yea we're gonna be dealing with some race conditions around this & the rest of the app init, fun

File: src/services/notificationService/registerNotifications.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      notification.finish(PushNotificationIOS.FetchResult.NoData);
+    },
+
+    onAction(notification) {
+      console.log('ACTION:', notification.action);
+      console.log('NOTIFICATION:', notification);
+    },
+
+    onRegistrationError(err) {
+      if (err.code !== 3010) {
```
Comment (garrettjoecox) [inline src/services/notificationService/registerNotifications.ts:22]: What is 3010?

**Review by garrettjoecox — APPROVED** [review]
Looks good!

File: src/services/notificationService/registerNotifications.ts (hunk 3 above)
Comment (s-fletcher) [reply src/services/notificationService/registerNotifications.ts:22]: That's the error when it's trying to register the app on a simulator. I'll add a comment for clarification.

**Pull request merged**